### ISSUE TITLE
Account for details

### ISSUE THREAD

**Issue by nicolasugrinovic** [facebookresearch/DeepSDF#8]
Hi ,
Thank you for sharing your great work. I did some quick experiments with ShapeNetCoreV1 but I cannot get to reconstruct "high frequency details" from the learned latent vector. As shown in the image GT vs. reconstruction. Are there any parameters I can change to get a more detailed reconstruction? 

![image](https://user-images.githubusercontent.com/25443240/61578217-bf9ca600-aaf3-11e9-81b4-572848f770d9.png)

**Comment by jjparkcv** [comment]
Hi, is this model part of training or testing data?

Either ways, your best bet would be to sample more aggressively near the surface, (i.e. lower variance in sampling in PreprocessMesh.cpp) and decrease the L2 regularization of the latent vector Z (code_reg_lambda).

Let me know how that works,
thanks.

**Comment by nicolasugrinovic** [comment]
Thank you, I will check that out. I am using test data. 

Another question, in your paper you say that the model you have should be able to represent a wide variety of shapes. Have you tried to learn one network parameters that works with all shape classes of the training set (at once)? I can see that you have a separate spec file for every shape class.

**Comment by nicolasugrinovic** [comment]
Hi @jjparkcv , 
I have tried to lower variance from default 0.005 to 0.001 and L2 reg. (CodeRegularizationLambda in specs) to 0.1e-4 and cannot see an improvement over the details.  Also, I  lowered ClampingDistance from 0.1 to 0.06, but could not appreciate more details. Could you advice for a range of values I should use?
Thanks!

**Comment by nicolasugrinovic** [comment]
Hi, thank you for your response. I have trained with parameters mentioned in the paper:

```
L2 reg = 1e-4
Truncation distance (ClampingDistance) = 0.01
Learning rate for decoder= 1e-5*B
Learning rate - latent = 1e-3
Variance in PreprocessMesh.cpp = 0.05
training epochs = 1160
```

However, I get the following results, were the left is GT and the right is reconstructed from latent code. The model is part of training.
![Selection_374](https://user-images.githubusercontent.com/25443240/62305561-7c8deb80-b480-11e9-9aaa-e8174e5b8785.png)

This next figure is from the paper and you get better details. Could you give me lights on what could I be doing wrong? 
![Selection_375](https://user-images.githubusercontent.com/25443240/62305597-8f082500-b480-11e9-8fbe-e2d3e16c668f.png)

**Comment by jjparkcv** [comment]
Hi Nicola, for high frequency details you need aggressive sampling near the surface. I notice that you used high variance, which means less aggressive surface sampling. I recommend using lower variance for SDF sample generation.

**Comment by nicolasugrinovic** [comment]
Hi Jeong, 
I have lowered the variance to `Variance in PreprocessMesh.cpp = 0.005
` and I have also trained another time with no regularization term. Still can't have good details in the reconstruction. 

What range of variance would you recommend?

![image](https://user-images.githubusercontent.com/25443240/62367510-2e352700-b52a-11e9-9fdb-af64c9a7b508.png)

**Comment by jjparkcv** [comment]
HI kadabra,

The result does not seem correct. Could you visualize the SDF samples
you're using? it seems like there's something wrong with the sampling.

**Comment by jjparkcv** [comment]
Hi Kadabra, I actually think I know what's going on.

When publishing the code we turned off the test that tells you if you're
mesh is valid (watertight) and instead provided a list of valid meshes in
shapenet, and the car class is not included.

So you might first want to try running the code in the classes that we
provide the list of valid meshes, and you might want to wait until we
publish a fix to revive the testing.

**Comment by nicolasugrinovic** [comment]
Hi Jeong, 
I have retrained with a much lower variance and the following parameters, but get the result shown in the picture, which is very similar to the one above. (All these trying to overfit with a dataset with 24 examples, inference done over training data). 

```
L2 reg = None
Truncation distance (ClampingDistance) = 0.01
Learning rate for decoder= 1e-5*B
Learning rate - latent = 1e-3
Variance in PreprocessMesh.cpp = 0.0002
training epochs = 120000
"SamplesPerScene" = 16384
"ScenesPerBatch" = 12
```
I have the network specs with defaults provided by your code:
`	"NetworkSpecs" : {
		"dims" : [ 512, 512, 512, 512, 512, 512, 512, 512 ],
		"dropout" : [0, 1, 2, 3, 4, 5, 6, 7],
		"dropout_prob" : 0.2,
		"norm_layers" : [0, 1, 2, 3, 4, 5, 6, 7],
		"latent_in" : [4],
		"xyz_in_all" : false,
		"use_tanh" : false,
		"latent_dropout" : false,
		"weight_norm" : true
		},
` 

![image](https://user-images.githubusercontent.com/25443240/62461016-f1646c80-b783-11e9-8b38-6050dbc7a60b.png)

**Comment by jjparkcv** [comment]
Hi Kadabra, we found out some issue.

When publishing the code we turned off the test that tells you if you're mesh is valid (watertight) and instead provided a list of valid meshes in some of the ShapeNet classes, and the car class is not included. 

So, it's very likely that there are many non-valid meshes being used during training for your case. We are in the process of re-introducing the valid-mesh testing code in the repo.

So you might first want to try running the code in the classes that we provide the list of valid meshes (e.g. chair), and you might want to wait until we publish a fix to revive the testing.

**Comment by nicolasugrinovic** [comment]
Hi, thank you for the answer! I understand. 
I have made the same training as before, just changing the class from cars to sofas. I get the following result, still can't get high freq. details. I don't know what I am doing wrong, would appreciate some insight. 

![Selection_181](https://user-images.githubusercontent.com/25443240/62560864-3bca1400-b87e-11e9-9424-269daa7613df.png)

**Comment by moshanATucsd** [comment]
Hi, I also want to try out the car class. @nicolasugrinovic could you kindly provide your splits files for training and testing?

**Comment by moshanATucsd** [comment]
Hi @jjparkcv, I am wondering what if we directly use the non-valid meshes for training, are we still able to learn some reasonable shape or not?

**Comment by StephenNG59** [comment]
> Hi Kadabra, we found out some issue.
> 
> When publishing the code we turned off the test that tells you if your mesh is valid (watertight) and instead provided a list of valid meshes in some of the ShapeNet classes, and the car class is not included.
> 
> So, it's very likely that there are many non-valid meshes being used during training for your case. We are in the process of re-introducing the valid-mesh testing code in the repo.
> 
> So you might first want to try running the code in the classes that we provide the list of valid meshes (e.g. chair), and you might want to wait until we publish a fix to revive the testing.

Hi, @jjparkcv. Is there a list of valid meshes somewhere but i couldn't find?

**Comment by gattia** [comment]
@nicolasugrinovic - did the improved code to test if the mesh was valid improve things for you?
